### ISSUE TITLE
Logs still written to stdout even when alternate driver specified

### ISSUE THREAD

**Issue by chiefy** [docker/compose#2652]
I'm not really sure when this happened, but I just noticed it on some of our prod boxes. When you run `docker compose` and one of your containers uses an alternate log format (in our case "gelf"), logs still get pushed to stdout. 

```
❯ docker --version
Docker version 1.9.1, build a34a1d5
❯ docker-compose --version
docker-compose version 1.5.2, build unknown
```

```
...
web:
  image: nginx
  log_driver: "gelf"
  log_opt:
    gelf-address: udp://172.0.1.2:12201
    gelf-tag: "test"
  links:
    - "logstash"
  ports:
    - "80:80"
```

```
logstash_1 | {:timestamp=>"2016-01-13T21:11:36.707000+0000", :message=>"Starting gelf listener", :address=>"0.0.0.0:12201", :level=>:info, :file=>"logstash/inputs/gelf.rb", :line=>"87", :method=>"udp_listener"}
web_1      | 192.168.99.1 - - [13/Jan/2016:21:11:41 +0000] "GET / HTTP/1.1" 304 0 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.106 Safari/537.36" "-"
logstash_1 | {:timestamp=>"2016-01-13T21:11:41.575000+0000", :message=>"output received", :event=>{"version"=>"1.1", "host"=>"docker-host", "short_message"=>"192.168.99.1 - - [13/Jan/2016:21:11:41 +0000] \"GET / HTTP/1.1\" 304 0 \"-\" \"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.106 Safari/537.36\" \"-\"", "level"=>6, "facility"=>"", "@version"=>"1", "@timestamp"=>"2016-01-13T21:11:41.000Z", "source_host"=>"172.17.0.1", "message"=>"", "command"=>"nginx -g daemon off;", "container_id"=>"91dd9e798b5138573d603618efad8c4cfbcf27097300b214da91927eb179b622", "container_name"=>"dclogtest_web_1", "created"=>"2016-01-13T21:11:25.232913722Z", "image_id"=>"407195ab8b072ce8b237f664b8d032704e8047b8e64139cc2e017163f7161647", "image_name"=>"nginx", "tag"=>"test", "type"=>"docker-log"}, :level=>:debug, :file=>"(eval)", :line=>"21", :method=>"output_func"}
```

**Comment by chiefy** [comment]
Can anyone from the team comment on this? It seems like unexpected behavior?

**Comment by dnephin** [comment]
It is unexpected. Do you see the same logs in logstash?

Are you able to reproduce this every time, or is it occasionally ?

**Comment by chiefy** [comment]
@dnephin yes, the log driver seems to be working fine, other than outputting to stdout as well. I can reproduce this every time.

**Comment by hussambuttalayacare** [comment]
@dnephin I am running into the same issue as chiefy but my docker-compose version is 1.6.2. I do not see any logs coming in to logstash but I do see them in stdout for the containers. I have two servers where I am facing this issue.
If I downgrade the  docker-compose to 1.5.2 on these servers though, stop and start the containers I see the logs coming in to my logstash. Any other servers with 1.5.2 have no issues pushing the logs to logstash.

`$ docker --version`
`Docker version 1.9.1, build a34a1d5`
`$ docker-compose --version`
`docker-compose version 1.6.2, build 4d72027`

**Comment by dnephin** [comment]
Sounds like the container didn't get the log config. Can you include the output of `docker-compose config` and a `docker inspect` for the container it created?

**Comment by hussambuttalayacare** [comment]
You are correct, I do no see the container picking up the log config for some reason but I am not sure why. The output for the docker-compose config on the container:
```
networks: {}
services:
  nginx:
    dns:
    - 192.168.2.25
    dns_search:
    - XXXXXXX
    image: nginx
    ports:
    - 80:80
    - 443:443
    volumes:
    - /srv/apps/ssl:/etc/nginx/ssl:rw
    volumes_from:
    - service:app:rw
version: '1'
volumes: {}
```

And for docker inspect
```
[
{
    "Id": "8e2ca394573da82cf937b3636a54cb28b765308f3c9a6157470cb342a445b427",
    "Created": "2016-03-09T20:55:58.066892515Z",
    "Path": "/usr/local/bin/update.sh",
    "Args": [
        "/run.sh"
    ],
    "State": {
        "Status": "running",
        "Running": true,
        "Paused": false,
        "Restarting": false,
        "OOMKilled": false,
        "Dead": false,
        "Pid": 27326,
        "ExitCode": 0,
        "Error": "",
        "StartedAt": "2016-03-09T20:55:58.388039864Z",
        "FinishedAt": "0001-01-01T00:00:00Z"
    },
    "Image": "d808719ea34b787496ed41776a2d365edc48edbe2fc7f9faeeda59bd312d7234",
    "ResolvConfPath": "/var/lib/docker/containers/8e2ca394573da82cf937b3636a54cb28b765308f3c9a6157470cb342a445b427/resolv.conf",
    "HostnamePath": "/var/lib/docker/containers/8e2ca394573da82cf937b3636a54cb28b765308f3c9a6157470cb342a445b427/hostname",
    "HostsPath": "/var/lib/docker/containers/8e2ca394573da82cf937b3636a54cb28b765308f3c9a6157470cb342a445b427/hosts",
    "LogPath": "/var/lib/docker/containers/8e2ca394573da82cf937b3636a54cb28b765308f3c9a6157470cb342a445b427/8e2ca394573da82cf937b3636a54cb28b765308f3c9a6157470cb342a445b427-json.log",
    "Name": "/nginx_1",
    "RestartCount": 0,
    "Driver": "aufs",
    "ExecDriver": "native-0.2",
    "MountLabel": "",
    "ProcessLabel": "",
    "AppArmorProfile": "",
    "ExecIDs": null,
    "HostConfig": {
        "Binds": [
            "/srv/apps/ssl:/etc/nginx/ssl:rw"
        ],
        "ContainerIDFile": "",
        "LxcConf": null,
        "Memory": 0,
        "MemoryReservation": 0,
        "MemorySwap": 0,
        "KernelMemory": 0,
        "CpuShares": 0,
        "CpuPeriod": 0,
        "CpusetCpus": "",
        "CpusetMems": "",
        "CpuQuota": 0,
        "BlkioWeight": 0,
        "OomKillDisable": false,
        "MemorySwappiness": null,
        "Privileged": false,
        "PortBindings": {
            "443/tcp": [
                {
                    "HostIp": "",
                    "HostPort": "443"
                }
            ],
            "80/tcp": [
                {
                    "HostIp": "",
                    "HostPort": "80"
                }
            ]
        },
        "Links": null,
        "PublishAllPorts": false,
        "Dns": [
            "192.168.2.25"
        ],
        "DnsOptions": null,
        "DnsSearch": [
            "XXXXXXX"
        ],
        "ExtraHosts": null,
        "VolumesFrom": [
            "8b39c0168fbdd51629400989ad828c0f38b9bcf4c2197564add67ec53df5b56c:rw"
        ],
        "Devices": null,
        "NetworkMode": "default",
        "IpcMode": "",
        "PidMode": "",
        "UTSMode": "",
        "CapAdd": null,
        "CapDrop": null,
        "GroupAdd": null,
        "RestartPolicy": {
            "Name": "",
            "MaximumRetryCount": 0
        },
        "SecurityOpt": null,
        "ReadonlyRootfs": false,
        "Ulimits": null,
        "LogConfig": {
            "Type": "json-file",
            "Config": {}
        },
        "CgroupParent": "",
        "ConsoleSize": [
            0,
            0
        ],
        "VolumeDriver": ""
    },
    "GraphDriver": {
        "Name": "aufs",
        "Data": null
    },
    "Mounts": [
        {
            "Source": "/srv/apps/ssl",
            "Destination": "/etc/nginx/ssl",
            "Mode": "rw",
            "RW": true
        },
        {
            "Name": "6c4208b28d74db6a3538b16836dfd3587a4ac6e00b66b95b6b864ebd4e90c720",
            "Source": "/var/lib/docker/volumes/6c4208b28d74db6a3538b16836dfd3587a4ac6e00b66b95b6b864ebd4e90c720/_data",
            "Destination": "/data/docroot",
            "Driver": "local",
            "Mode": "",
            "RW": true
        }
    ],
    "Config": {
        "Hostname": "8e2ca394573d",
        "Domainname": "",
        "User": "",
        "AttachStdin": false,
        "AttachStdout": false,
        "AttachStderr": false,
        "ExposedPorts": {
            "443/tcp": {},
            "80/tcp": {}
        },
        "Tty": false,
        "OpenStdin": false,
        "StdinOnce": false,
        "Cmd": [
            "/run.sh"
        ],
        "Image": "nginx",
        "Volumes": {
            "/etc/nginx/ssl": {}
        },
        "WorkingDir": "",
        "Entrypoint": [
            "/usr/local/bin/update.sh"
        ],
        "OnBuild": null,
        "Labels": {
            "com.docker.compose.config-hash": "9b3d0eea470f5e6b7ed3e4c427200e40ab1970b172744fe7addcc9dd9cf80a44",
            "com.docker.compose.container-number": "1",
            "com.docker.compose.oneoff": "False",
            "com.docker.compose.project": "foo",
            "com.docker.compose.service": "nginx",
            "com.docker.compose.version": "1.6.2"
        }
    },
    "NetworkSettings": {
        "Bridge": "",
        "SandboxID": "0f1c60d431c90dd588b0779cc43104f2616cf1351460896e2ccece9f88f0ba2c",
        "HairpinMode": false,
        "LinkLocalIPv6Address": "",
        "LinkLocalIPv6PrefixLen": 0,
        "Ports": {
            "443/tcp": [
                {
                    "HostIp": "0.0.0.0",
                    "HostPort": "443"
                }
            ],
            "80/tcp": [
                {
                    "HostIp": "0.0.0.0",
                    "HostPort": "80"
                }
            ]
        },
        "SandboxKey": "/var/run/docker/netns/0f1c60d431c9",
        "SecondaryIPAddresses": null,
        "SecondaryIPv6Addresses": null,
        "EndpointID": "827649ddac3f94c1b0fb06a4761243d5bea06172ce089199f819177eb4b74e9c",
        "Gateway": "172.17.0.1",
        "GlobalIPv6Address": "",
        "GlobalIPv6PrefixLen": 0,
        "IPAddress": "172.17.0.7",
        "IPPrefixLen": 16,
        "IPv6Gateway": "",
        "MacAddress": "",
        "Networks": {
            "bridge": {
                "EndpointID": "827649ddac3f94c1b0fb06a4761243d5bea06172ce089199f819177eb4b74e9c",
                "Gateway": "172.17.0.1",
                "IPAddress": "172.17.0.7",
                "IPPrefixLen": 16,
                "IPv6Gateway": "",
                "GlobalIPv6Address": "",
                "GlobalIPv6PrefixLen": 0,
                "MacAddress": ""
            }
        }
    }
}
]
```

**Comment by dnephin** [comment]
I think you're hitting the bug fixed by #3052. The fix is in master and will be in 1.7.0. If you want to try a master build, you can download from https://dl.bintray.com/docker-compose/master/
